src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeUpdateService.java

```diff
@@ -31,24 +32,25 @@ public EnvelopeUpdateService(
         this.statusChangeValidator = statusChangeValidator;
     }
 
-    /**
-     * Changes the status of envelope to DOC_CONSUMED.
-     *
-     * @param envelopeId  ID of the envelope to update
-     * @param serviceName Name of the service that requests to mark the envelope as consumed
-     */
-    public void markAsConsumed(UUID envelopeId, String serviceName) {
+    public void updateStatus(UUID envelopeId, NewStatus newStatus, String serviceName) {
+
         Envelope envelope =
             envelopeRepo
                 .findById(envelopeId)
                 .orElseThrow(() -> new EnvelopeNotFoundException());
 
         accessService.assertCanUpdate(envelope.getJurisdiction(), serviceName);
-        statusChangeValidator.assertCanUpdate(envelope.getStatus(), Status.CONSUMED);
 
-        envelope.setStatus(Status.CONSUMED);
-        envelopeRepo.save(envelope);
+        if (newStatus == NewStatus.CONSUMED) {
```

Comment:
We could avoid if else ?

```

if (newStatus != NewStatus.CONSUMED) {
 throw new IllegalStateException("No code for handling status: " + newStatus);
}

statusChangeValidator.assertCanUpdate(envelope.getStatus(), Status.CONSUMED);

envelope.setStatus(Status.CONSUMED);
envelopeRepo.save(envelope);

eventRepo.save(new ProcessEvent(envelope.getContainer(), envelope.getZipFileName(), Event.DOC_CONSUMED));

```

Reply:
Not sure if throwing is IllegalStateException  appropriate here
> Signals that a method has been invoked at an illegal or inappropriate time. In other words, the Java environment or Java application is not in an appropriate state for the requested operation.

Reply:
I consider `if-else` better than `if` in general.
It's more explicit.